common/common.go

```diff
@@ -512,20 +519,31 @@ func RunBuild(verbose bool, dir, imageName, dockerfile string, buildArgs []strin
 	go func(done chan<- error) {
 		var dockerBuildCmdArgs []string
 		// Depending whether architecture list is passed or not trigger docker buildx or docker build accordingly
-		var mappedArchitectures []string
+
 		if arch, ok := ShapeMap[shape]; ok {
+			var mappedArchitectures []string
 			mappedArchitectures = append(mappedArchitectures, arch...)
-			err := initializeContainerBuilder(containerEngineType, mappedArchitectures)
-			if err != nil {
-				done <- err
-				return
+			var hostedPlatform = runtime.GOARCH
+			if platform, ok := TargetPlatformMap[shape]; ok {
+				targetPlatform := strings.Join(platform," ")
+				if targetPlatform != hostedPlatform {
+					fmt.Println("TargetedPlatform and hostPlatform are not same")
+					if containerEngineType == ContainerEngineType {
```

Comment:
Change the variable name to reflect docker type.
Please move this condition to the method

Reply:
done